src/libstd/io/stdio.rs

```diff
@@ -81,11 +81,17 @@ impl Read for StdinRaw {
 }
 impl Write for StdoutRaw {
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> { self.0.write(buf) }
+    #[cfg(not(target_os = "redox"))]
     fn flush(&mut self) -> io::Result<()> { Ok(()) }
+    #[cfg(target_os = "redox")]
+    fn flush(&mut self) -> io::Result<()> { self.0.flush() }
```

Comment:
On Redox, the `stdio` needs to be flushed when in raw mode. This perhaps could be removed by adjusting programs that enter raw mode